### PR TITLE
Rework Find-in-Files Follow-current-doc

### LINKED ISSUE #15908: [BUG] Follow current doc. option should coordinate better with the Directory input field
### Steps to Reproduce the Issue
1.  Open Find in Files
2.  Check **Follow current doc.** checkbox (The directory displayed will be updated to the current one if it isn't already)
3.  Manually edit the directory to a different one
4.  Click **Find All**

Then the search is executed in the given directory despite the **Follow current doc.** checkbox 

The checkbox should either:
- gray out the Directory field
- be automatically unchecked if the Directory field is modified

### Debug Information
Notepad++ v8.7.1   (64-bit)

### PR DESCRIPTION
**Pull request by alankilborn**
Fix #15908 

---

Notes / questions:

* For the hover-text on the `<<` button, I went with a "sentence case" version of the string (for consistency; such tips in the UI are this way), rather than the specified "proper case" version.
* (Outside the spec's scope...) I added hover-text for the `...` button as well, as I found it strange to have two buttons very near each other, with symbols (not text) as captions on both, and leaving only ONE of them with hover-text explaining functionality.
* I reused the same number resource number (1661) for the new `<<` button as the old/removed `Follow current doc.` checkbox.  There is no translation for `<<` (obviously). Will this cause problem if translators leave the translation for 1661 in? Should I use a new number instead of 1661 to avoid this potential?
* New pref. setting `Fill Find in Files Directory Field Based On Active Document` is checkmarked by default (this was _my_ design decision); removed `Follow current doc.` setting was NOT checkmarked by default.  Should this stand?
* For changes relating specifically to this PR, `_isFolderFollowDoc` is relevant to it, but I also noticed `_isFilterFollowDoc` nearby in the code.  Further investigation showed this is a hidden-from-user (no UI control) setting in `config.xml` (`"fifFilterFollowsDoc"`).  Should this be removed from hidden status and given its own setting in the _Preferences_ > _Searching_ ?

### CODE REVIEW

File: PowerEditor/src/WinControls/Preference/preference.rc [hunk 1]
```diff
@@ -384,17 +384,18 @@ IDD_PREFERENCE_SUB_SEARCHING DIALOGEX 115, 10, 460, 205
...
-    CONTROL         "Find dialog remains open after search that outputs to results window", IDC_CHECK_FINDDLG_ALWAYS_VISIBLE, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 89, 350, 10
-    CONTROL         "Confirm Replace All in All Opened Documents", IDC_CHECK_CONFIRMREPLOPENDOCS, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 104, 350, 10
-    CONTROL         "Replace: Don't move to the following occurrence", IDC_CHECK_REPLACEANDSTOP, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 119, 350, 10
-    CONTROL         "Search Result window: show only one entry per found line if possible", IDC_CHECK_SHOWONCEPERFOUNDLINE, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 134, 350, 10
+    CONTROL         "Fill Find in Files Directory Field Based On Active Document", IDC_CHECK_FILL_DIR_FIELD_FROM_ACTIVE_DOC, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 75, 350, 10
+    CONTROL         "Use Monospaced font in Find dialog (Need to restart Notepad++)", IDC_CHECK_MONOSPACEDFONT_FINDDLG, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 134, 350, 10
```
Comment (notdodgeball) [inline PowerEditor/src/WinControls/Preference/preference.rc:394]: A little typo, it should be 104 I think.

**Comment by donho** [comment]
> For the hover-text on the `<<` button, I went with a "sentence case" version of the string (for consistency; such tips in the UI are this way), rather than the specified "proper case" version.

Yes, sure. The current format of tip is OK for me.

> (Outside the spec's scope...) I added hover-text for the `...` button as well, as I found it strange to have two buttons very near each other, with symbols (not text) as captions on both, and leaving only ONE of them with hover-text explaining functionality.

It's not necessary. `...` is clear for the users. For the worst case, they will click on it, and see what it is exactly.

> I reused the same number resource number (1661) for the new `<<` button as the old/removed `Follow current doc.` checkbox.  There is no translation for `<<` (obviously). Will this cause problem if translators leave the translation for 1661 in? Should I use a new number instead of 1661 to avoid this potential?

Normally we should always find a new number to avoid the translation colision.
But it's OK for this time. I'll remove 1161 entry after the PR being merged.

> New pref. setting `Fill Find in Files Directory Field Based On Active Document` is checkmarked by default (this was _my_ design decision); removed `Follow current doc.` setting was NOT checkmarked by default.  Should this stand?

No. Please let's keep unchecked as the old default behaviour.

> For changes relating specifically to this PR, `_isFolderFollowDoc` is relevant to it, but I also noticed `_isFilterFollowDoc` nearby in the code.  Further investigation showed this is a hidden-from-user (no UI control) setting in `config.xml` (`"fifFilterFollowsDoc"`).  Should this be removed from hidden status and given its own setting in the _Preferences_ > _Searching_ ?

It's a hidden argument added in v5.4:
***Add an (hidden) option to make the "Filters" and "Directory" fields follow the current opened document in "Find in files" dialog. The new parameters are located in config.xml and are called: "fifFilterFollowsDoc" and "fifFolderFollowsDoc" (set to "yes"/"no")***
(ref: https://github.com/ArkadiuszMichalski/NPP_tools/blob/main/_change/change.md)
So no, just keep it hidden. BTW, it seems not being in Npp-User-Manual. @pryrt Do you confirm it?


One request - personally I prefer this way:

image

So please use the following code:
```
IDD_PREFERENCE_SUB_SEARCHING DIALOGEX 115, 10, 460, 205
STYLE DS_SETFONT | DS_FIXEDSYS | DS_CONTROL | WS_CHILD
FONT 8, "MS Shell Dlg", 0, 0, 0x1
BEGIN
    GROUPBOX        "When Find Dialog is Invoked",IDD_FILL_FIND_FIELD_GRP_STATIC,31,4,323,85,BS_CENTER
    RTEXT           "Minimum Size for Auto-Checking ""In selection"":",IDC_INSELECTION_THRESHOLD_STATIC,37,16,210,8
    PUSHBUTTON      "?",IDC_INSELECTION_THRESH_QUESTION_BUTTON,283,14,16,14,NOT WS_TABSTOP
    EDITTEXT        IDC_INSELECTION_THRESHOLD_EDIT,250,15,27,12,ES_CENTER | ES_NUMBER
    CONTROL         "Fill Find Field with Selected Text",IDC_CHECK_FILL_FIND_FIELD_WITH_SELECTED,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,59,300,10
    CONTROL         "Select Word Under Caret when Nothing Selected",IDC_CHECK_FILL_FIND_FIELD_SELECT_CARET,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,52,73,275,10
    CONTROL         "Fill Find in Files Directory Field Based On Active Document",IDC_CHECK_FILL_DIR_FIELD_FROM_ACTIVE_DOC,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,44,311,8
    CONTROL         "Use Monospaced font in Find dialog (Need to restart Notepad++)",IDC_CHECK_MONOSPACEDFONT_FINDDLG,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,104,350,10
    CONTROL         "Find dialog remains open after search that outputs to results window",IDC_CHECK_FINDDLG_ALWAYS_VISIBLE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,119,350,10
    CONTROL         "Confirm Replace All in All Opened Documents",IDC_CHECK_CONFIRMREPLOPENDOCS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,134,350,10
    CONTROL         "Replace: Don't move to the following occurrence",IDC_CHECK_REPLACEANDSTOP,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,149,350,10
    CONTROL         "Search Result window: show only one entry per found line if possible",IDC_CHECK_SHOWONCEPERFOUNDLINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,164,350,10
END
```

**Review by donho — CHANGES_REQUESTED** [review]
Please consider the [suggestions](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15938#issuecomment-2543059389).

**Comment by alankilborn** [comment]
@donho 

>BTW, it (fifFilterFollowsDoc) seems not being in Npp-User-Manual. @pryrt Do you confirm it?

It is there, or will be there soon.
When I was investigating the code change for this PR, I saw it and gave @pryrt the "heads up" about it, and he said he would put it in the UM.

**Comment by alankilborn** [comment]
@donho

> It's a hidden argument added in v5.4:
> _**Add an (hidden) option to make the "Filters" and "Directory" fields follow the current opened document in "Find in files" dialog. The new parameters are located in config.xml and are called: "fifFilterFollowsDoc" and "fifFolderFollowsDoc" (set to "yes"/"no")**_

> So no, just keep it hidden.

In my opinion, it doesn't make a lot of sense to have one available as a preference checkbox choice, and one a super-secret hidden item (it's inconsistent, a big reason for changes I've suggested/implemented recently)... but OK, I will keep **fifFilterFollowsDoc** hidden from user.

**Comment by alankilborn** [comment]
@donho 

>It's (hover-tip-text) not necessary. 
`...` is clear for the users. 

Again, it plays to "inconsistency".
Here we've got two buttons, side by side:  `...` and `<<`.
The novice user sees these buttons and does not know what they do.
He thinks "let's see if there is a hover-text tip" and hovers mouse over `<<` and sees the tip (he thinks "nice!").
He moves mouse to point at `...` button and, hmm, no help tip (he thinks "darn!")

>For the worst case, they will click on it, and see what it is exactly.

This is also true for new button `<<`.

---

But OK, I will remove hover-text-tip for `...` button.

**Comment by alankilborn** [comment]
@donho

>Please consider the suggestions

Everything is implementable, so I'll go ahead and implement them.
Thank you for reading my "Notes" and responding to each point.

**Comment by pryrt** [comment]
> it seems not being in Npp-User-Manual. @pryrt Do you confirm it?

I recently added it to https://npp-user-manual.org/docs/preferences/#preferences-for-advanced-users , per @alankilborn 's prompting.

**Comment by donho** [comment]
@pryrt 
Thank you!

**Pull request closed without merging**